include/clspv/Compiler.h

```diff
@@ -12,7 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#ifndef CLSPV_INCLUDE_CLSPV_COMPILER_H_
+#define CLSPV_INCLUDE_CLSPV_COMPILER_H_
+
+#include <string>
+#include <vector>
+
+#include "clspv/DescriptorMap.h"
+
 namespace clspv {
+struct Result {
```

Comment:
Have you considered using `StringError` and friends?

Reply:
Right now the API doesn't expose LLVM headers and I'd prefer to keep it that way.